component/src/main/java/org/wso2/extension/siddhi/execution/rdbms/QueryStreamProcessor.java

```diff
@@ -0,0 +1,253 @@
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "query",
+                        description = "The select query that needs to be performed",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "attribute.definition.list",
+                        description = "Comma separated list of `<AttributeName AttributeType>`. " +
+                                "It is expected that the SQL query will return the attributes in order, as in if one " +
+                                "attribute is defined here, the SQL query should return one column result set, " +
+                                "if more than one column is returned then the first column will be processed. The " +
+                                "Siddhi data types supported will be `STRING`, `INT`, `LONG`, `DOUBLE`, `FLOAT`, " +
+                                "`BOOL`. \n Mapping of the Siddhi data type to database data type can be done as " +
+                                "follows, \n" +
+                                "*Siddhi Datatype*->*Datasource Datatype*\n" +
+                                "`STRING`->`CHAR`,`VARCHAR`,`LONGVARCHAR`\n" +
+                                "`INT`&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;->`INTEGER`\n" +
+                                "`LONG`&nbsp;&nbsp;&nbsp;&nbsp;->`BIGINT`\n" +
+                                "`DOUBLE`->`DOUBLE`\n" +
+                                "`FLOAT`&nbsp;&nbsp;&nbsp;->`REAL`\n" +
+                                "`BOOL`&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;->`BIT`\n",
+                        type = DataType.STRING
+                )
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "custom",
+                        description = "The return attributes will be the ones defined in the parameter" +
+                                "`attribute.definition.list`.",
+                        type = DataType.STRING
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "from TriggerStream#rdbms:query('SAMPLE_DB', 'select * from " +
+                                "Transactions_Table', 'creditcardno string, country string, transaction string," +
+                                " amount int')  \n" +
+                                "insert into  recordStream;",
+                        description = "Events inserted into recordStream includes all records matched for the query " +
+                                "i.e a single event will be generated for each record retrieved from the datasource. " +
+                                "The event will include as additional attributes, the attributes defined in the " +
+                                "function(creditcardno, country, transaction, amount)."
```

Comment:
function? it should be "attribute.definition.list"

Reply:
Fixed in 663e63b